snapdtool/info_file.go

```diff
@@ -20,34 +20,53 @@
 package snapdtool
 
 import (
-	"bytes"
+	"bufio"
 	"fmt"
-	"io/ioutil"
+	"os"
+	"strings"
 )
 
 // SnapdVersionFromInfoFile returns snapd version read for the
 // given info" file, pointed by infoPath.
-// The format of the "info" file is a single line with "VERSION=..."
-// in it. The file is produced by mkversion.sh and normally installed
-// along snapd binary in /usr/lib/snapd.
-func SnapdVersionFromInfoFile(infoPath string) (string, error) {
-	content, err := ioutil.ReadFile(infoPath)
+// The format of the "info" file are lines with "KEY=VALUE" with the typical key
+// being just VERSION. The file is produced by mkversion.sh and normally
+// installed along snapd binary in /usr/lib/snapd.
+// Other typical keys in this file include SNAPD_APPARMOR_REEXEC, which
+// indicates whether or not the snapd-apparmor binary installed via the
+// traditional linux package of snapd supports re-exec into the version in the
+// snapd or core snaps.
+func SnapdVersionFromInfoFile(infoPath string) (version string, flags map[string]string, err error) {
```

Comment:
given we are changing this - as I mentioned - it's probably worth change the argument to be the dir, we don't have a use case for the filename itself != "info"

Reply:
done